### PR TITLE
Add a residual convolutional layer

### PR DESCRIPTION
**Pull request by devisperessutti**
This PR adds a residual convolutional block, as described in the following figure.

res-conv-2d-diagram

Batch normalisation is optional. Different values for `kernel_size` and `dilation` can be provided for each convolutional block.

### CODE REVIEW

File: eoflow/models/segmentation_base.py [hunk 1]
```diff
@@ -29,7 +31,8 @@
     'accuracy': lambda: tf.keras.metrics.CategoricalAccuracy(name='accuracy'),
     'iou': lambda: MeanIoU(default_max_classes=32),
     'precision': tf.keras.metrics.Precision,
-    'recall': tf.keras.metrics.Recall
+    'recall': tf.keras.metrics.Recall,
+    'mcc': lambda: MCCMetric()
```
Comment (veseln) [inline eoflow/models/segmentation_base.py:35]: Isn't this just 
...
`'mcc': MCCMetric`
instead of 
`'mcc': lambda: MCCMetric()  `

File: eoflow/models/layers.py [hunk 2]
```diff
@@ -166,6 +167,49 @@ def call(self, inputs, training=False):
...
+        self.convs = [Conv2D(filters,
+                             kernel_size=k,
+                             strides=strides,
+                             dilation=d,
+                             padding=padding,
+                             activation=activation,
+                             add_dropout=add_dropout,
+                             dropout_rate=dropout_rate,
+                             batch_normalization=batch_normalization,
+                             num_repetitions=2) for _, k, d in zip(range(num_parallel), kernel_list, dilation_list)]
```
Comment (veseln) [inline eoflow/models/layers.py:203]: Why the range since it is not used? 
Wouldn't 
```
for k, d in zip(kernel_list, dilation_list) 
```
work as well?

Comment (devisperessutti) [reply eoflow/models/layers.py]: yes, indeed. Changed as suggested

File: eoflow/models/segmentation_base.py (hunk 1 above)
Comment (devisperessutti) [reply eoflow/models/segmentation_base.py]: yes, this should do as well, changed
